### PR TITLE
[nccl-pg] Migrate to getCvar* functions for env variable checking

### PR DESCRIPTION
**Pull request by pavanbalaji**
Summary:
The getCvar* functions allow us to provide multiple environment variables for the same value.  This allows us to deprecate some variables in favor of others, while still allowing users to temporarily use the old variables for some time.

Test Plan: OSS CI

Reviewed By: fduwjj, XilunWu

Differential Revision: D51225487

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/distributed/c10d/Utils.hpp [hunk 1]
```diff
@@ -87,50 +85,99 @@ inline std::vector<std::string> split(
...
 }
 
-inline int parseEnvVarInt(const char* envVarName) {
-  char* stringValue = std::getenv(envVarName);
-  if (stringValue != nullptr) {
-    int val;
+inline std::string getCvarString(const std::vector<std::string>& env, const char *def) {
+  const char *ret = def;
+
+  if (env.size() == 0) {
```
Comment (Skylion007) [inline torch/csrc/distributed/c10d/Utils.hpp:91]:
```suggestion
  if (env.empty()) {
```

**Review by fduwjj — APPROVED** [review]
Please ensure all CI pass.

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 2]
```diff
@@ -39,6 +39,22 @@ namespace c10d {
...
+    "TORCH_NCCL_ENABLE_MONITORING"};
+std::vector<std::string> TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC = {
+    "TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC"};
+std::vector<std::string> TORCH_NCCL_AVOID_RECORD_STREAMS = {
+    "TORCH_NCCL_AVOID_RECORD_STREAMS"};
+std::vector<std::string> NCCL_USE_TENSOR_REGISTER_ALLOCATOR_HOOK = {
+    "NCCL_USE_TENSOR_REGISTER_ALLOCATOR_HOOK"};
+
 DebugInfoWriter::DebugInfoWriter(int rank) {
   const char* fileName = parseEnvVarString(
```
Comment (XilunWu) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:59]: need to replace this as well

**Review by XilunWu — CHANGES_REQUESTED** [review]
we can merge once CI signals are all green ;-) Hope you enjoyed the dev experience.

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp (hunk 2 above)
Comment (pavanbalaji) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp]: This patch does not rename the variables yet.  It simply introduces the new getCvar functions.  I have another commit that makes the actual changes to the env variables, which I have not pushed yet.

**Pull request closed without merging**